check.sh

```diff
@@ -37,6 +37,13 @@ if git status --porcelain | grep -q "requirements.txt"; then
     EXIT_STATUS=1
 fi
 
+python trio/_tests/check_type_completeness.py --overwrite-file || EXIT_STATUS=$?
```

Comment:
Just a note that we might want to check that type completeness won't use the platform it currently runs on for e.g. filtering out `if sys.version_info < ...` stuff. Not required though, this just popped into mind while seeing this.

Reply:
hmm. Maybe a reason to update the `check` environment to something newer than 3.8
I suppose a thorough check would run this on all supported versions, but that feels a bit overkill.

Reply:
but I don't fully know how pytest handles that stuff internally, which is what matters.

Reply:
I'm going to guess it handles this right and if it doesn't, we can fix then.